lambda_functions/execute_ssh_command_js/handler.js

```diff
@@ -34,41 +24,55 @@ function create_execution_string(event) {
      */
     let command_name = event['command'];
     delete event['command'];
-    let args = _(event.args)
+    let args = _(event)
         .transform((r, v, k) => r['--' + k.replace('_', '-')] = `'${v}'`)
         .toPairs().flatten().join(' ');
     return command_name + ' ' + args;
-}
+};
+
 // See https://hackernoon.com/you-should-use-ssm-parameter-store-over-lambda-env-variables-5197fc6ea45b
-// For a more rigorous way of loading data from SSM using KSM
+// For a more advanced way of loading data from SSM using KSM
 // Or look at https://github.com/n1ru4l/ssm-parameter-env
+exports.raijin_ssh_command = (event, context, callback) => {
+        let req = {
+                   Names: [hostkey, userkey, pkey],
+                   WithDecryption: true
+        };
+        let keys = ssm.getParameters(req).promise();
 
-module.exports.execute_ssh_command = (event, context, callback) => {
-    keys.then((data) => {
-        let params = {};
-        for (let p of data.Parameters) {
-            params[p.Name] = p.Value;
-        }
-
-        let ssh = new SSH({
-            host: params[hostkey],
-            user: params[userkey],
-            key: params[pkey],
+        keys.catch(function(err) {
+            console.log(err);
         });
-
-        let command = create_execution_string(event);
-//        let command = `execute_ingest --dea-module ${dea_module} --project ${project} --queue ${queue} --product ${product} --year ${year}`;
-        console.log(`Executing: ${command}`);
-        ssh.exec(command, {
-            out: function(stdout) {
-                console.log(stdout);
-                const response = { statusCode: 200, body: 'SSH command executed.' };
-                callback(null, response);
-            },
-            err: function(stderr) {
-                callback(`SSH Failed ${stderr}`);
+        keys.then((data) => {
+            let params = {};
+            for (let p of data.Parameters) {
+                 params[p.Name] = p.Value;
             }
-        }).start();
-    });
 
+            var ssh = new SSH({
+                               host: params[hostkey],
+                               user: params[userkey],
+                               key: params[pkey],
+                               });
+
+            console.log(`Host key: ${params[hostkey]}`);
+            console.log(`User key: ${params[userkey]}`);
+
+            let command = create_execution_string(event);
+            console.log(`Executing: ${command}`);
+
+	        ssh.exec(command, {
+                     exit: (code, stdout, stderr) => {
+                        console.log(`STDOUT: ${stdout}`);
+                        console.log(`STDERR: ${stderr}`);
+                     }
+                   })
+               .exec('exit 69', {
+                       exit: console.log
+                   })
+               .start({
+                   success: () => console.log('Successfully connected to Raijin system.'),
```

Comment:
Instead of `Raijin`, we should say change this to `${params[hostkey]}`, which will also require changing the quotes from `'` to ``` ` ```